File: JavaScript/Endpoint Examples/Multipart Payload/pdf-with-redacted-text-preview.js
/**
 * This request demonstrates how to apply text redaction previews to a PDF.
 * This is step 1 of 2 toward applying redactions to text.
 */
var axios = require('axios');
var FormData = require('form-data');
var fs = require('fs');

// Toggle deletion of sensitive files (default: false)
const DELETE_SENSITIVE_FILES = false;

// Create a new form data instance and append the PDF file and parameters to it
var data = new FormData();
data.append('file', fs.createReadStream('/path/to/file'));
var redaction_option_array = [];
var redaction_options1 = {
    "type": "preset",
    "value": "email",
};
var redaction_options2 = {
    "type": "regex",
    "value": "(\\+\\d{1,2}\\s)?\\(?\\d{3}\\)?[\\s.-]\\d{3}[\\s.-]\\d{4}",
};
var redaction_options3 = {
    "type": "literal",
    "value": "word",
};
redaction_option_array.push(redaction_options1);
redaction_option_array.push(redaction_options2);
redaction_option_array.push(redaction_options3);
data.append('redactions', JSON.stringify(redaction_option_array));
data.append('output', 'pdfrest_pdf_with_redacted_text_preview');

// define configuration options for axios request
var config = {
  method: 'post',
  maxBodyLength: Infinity, // set maximum length of the request body
  url: 'https://api.pdfrest.com/pdf-with-redacted-text-preview', 
  headers: { 
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', // Replace with your API key
    ...data.getHeaders() // set headers for the request
  },
  data : data // set the data to be sent with the request
};

// send request and handle response or error
axios(config)
.then(function (response) {
  console.log(JSON.stringify(response.data));

  // All files uploaded or generated are automatically deleted based on the 
  // File Retention Period as shown on https://pdfrest.com/pricing. 
  // For immediate deletion of files, particularly when sensitive data 
  // is involved, an explicit delete call can be made to the API.
  //
  // The following code is an optional step to delete sensitive files
  // (unredacted, unencrypted, unrestricted, or unwatermarked) from pdfRest servers.

  var body = response.data;
  var input_id = body.inputId;
  var preview_id = body.outputId;
  // IMPORTANT: Do not delete the preview_id (the preview PDF) file until after the redaction is applied
  // with the /pdf-with-redacted-text-applied endpoint.
  var delete_config = {
    method: 'post',
    maxBodyLength: Infinity,
    url: 'https://api.pdfrest.com/delete',
    headers: {
      'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
      'Content-Type': 'application/json'
    },
    data: { ids: input_id + ', ' + preview_id }
  };

  if (DELETE_SENSITIVE_FILES) {
    axios(delete_config)
      .then(function (delete_response) { console.log(JSON.stringify(delete_response.data)); })
      .catch(function (error) { console.log(error); });
  }
})
.catch(function (error) {
  console.log(error); 
});

// If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.js' sample.

File: Python/Endpoint Examples/Multipart Payload/pdf-with-redacted-text-preview.py
from requests_toolbelt import MultipartEncoder
import requests
import json

pdf_with_redacted_text_endpoint_url = 'https://api.pdfrest.com/pdf-with-redacted-text-preview'

redaction_options = [{
        "type": "preset",
        "value": "email",
    },
    {
        "type": "regex",
        "value": "(\\+\\d{1,2}\\s)?\\(?\\d{3}\\)?[\\s.-]\\d{3}[\\s.-]\\d{4}",
    },
    {
        "type": "literal",
        "value": "word",
    }]

mp_encoder_redactedtextPDF = MultipartEncoder(
    fields={
        'file': ('file_name.pdf', open('/path/to/file', 'rb'), 'application/pdf'),
        'redactions': json.dumps(redaction_options),
        'output' : 'example_out'
    }
)

headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_redactedtextPDF.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to pdf-with-redacted-text endpoint...")
response = requests.post(pdf_with_redacted_text_endpoint_url, data=mp_encoder_redactedtextPDF, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)

# All files uploaded or generated are automatically deleted based on the 
# File Retention Period as shown on https://pdfrest.com/pricing. 
# For immediate deletion of files, particularly when sensitive data 
# is involved, an explicit delete call can be made to the API.
#
# The following code is an optional step to delete sensitive files
# (unredacted, unencrypted, unrestricted, or unwatermarked) from pdfRest servers.

# IMPORTANT: Do not delete the outputId (the preview PDF) file until after the redaction is applied
# with the /pdf-with-redacted-text-applied endpoint.
delete_data = { "ids": response_json['inputId'] + ", " + response_json['outputId'] }
delete_response = requests.post(url='https://api.pdfrest.com/delete',
                data=json.dumps(delete_data),
                headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})
print("Delete response status code: " + str(delete_response.status_code))
print(delete_response.text if not delete_response.ok else json.dumps(delete_response.json(), indent = 2))
